provider/pkg/overlays/examples/callbackFunction.md

```diff
@@ -0,0 +1,216 @@
+A CallbackFunction is a special type of `aws.lambda.Function` that can be created out of an actual JavaScript function instance. The Pulumi compiler and runtime work in tandem to extract your function, package it up along with its dependencies, upload the package to AWS Lambda, and configure the resulting AWS Lambda resources automatically.
+
+The JavaScript function may capture references to other variables in the surrounding code, including other resources and even imported modules. The Pulumi compiler figures out how to serialize the resulting closure as it uploads and configures the AWS Lambda. This works even if you are composing multiple functions together.
+
+See [Function Serialization](https://www.pulumi.com/docs/concepts/inputs-outputs/function-serialization/) for additional details on this process.
+
+If no IAM Role is specified, CallbackFunction will automatically use the following managed policies:
```

Comment:
Is this important information to have upfront, or perhaps reference section at the bottom?